components/range_cache_memory_engine/src/background.rs

```diff
@@ -956,34 +970,90 @@ impl Runnable for BackgroundRunner {
                                     Err(e) => {
                                         error!("creating rocksdb iterator failed"; "cf" => cf, "err" => %e);
-                                        return false;
+                                        return None;
                                     }
                                 }
                             }
-                            true
+
+                            // gc the range
+                            let tso_timeout = std::cmp::min(gc_interval, Duration::from_secs(5));
+                            let now = match block_on_timeout(pd_client.get_tso(), tso_timeout) {
+                                Ok(Ok(ts)) => ts,
+                                err => {
+                                    error!(
+                                        "get timestamp failed, skip gc loaded range";
+                                        "timeout_duration" => ?tso_timeout,
+                                        "error" => ?err,
+                                    );
+                                    // Get timestamp fail so don't do gc.
+                                    return Some(0);
+                                }
+                            };
+
+                            let safe_point = (|| {
+                                fail::fail_point!("in_memory_engine_safe_point_in_loading", |t| {
+                                    t.unwrap().parse::<u64>().unwrap()
+                                });
+
+                                let safe_point = now
+                                    .physical()
+                                    .saturating_sub(gc_interval.as_millis() as u64);
+                                TimeStamp::compose(safe_point, 0).into_inner()
+                            })();
+
+                            let write_cf_handle = skiplist_engine.cf_handle(CF_WRITE);
+                            let default_cf_handle = skiplist_engine.cf_handle(CF_DEFAULT);
+                            let mut filter = Filter::new(
+                                safe_point,
+                                u64::MAX,
+                                default_cf_handle,
+                                write_cf_handle.clone(),
+                            );
+
+                            let mut iter = write_cf_handle.iterator();
+                            let guard = &epoch::pin();
+                            let (start_key, end_key) = encode_key_for_boundary_with_mvcc(&range);
+                            iter.seek(&start_key, guard);
+                            while iter.valid() && iter.key() < &end_key {
+                                let k = iter.key();
+                                let v = iter.value();
+                                if let Err(e) = filter.filter(k.as_bytes(), v.as_bytes()) {
+                                    warn!(
+                                        "Something Wrong in memory engine GC";
```

Comment:
```suggestion
                                        "something wrong in memory engine GC";
```

Reply:
How about evicting the range if it occurs error?

Reply:
It follows what what compaction filter does.